File: src/main/java/GUIs/MainGui.java
package GUIs;

import java.io.IOException;
import java.io.InputStream;
import java.net.URISyntaxException;
import javafx.application.Application;
import javafx.fxml.FXMLLoader;
import javafx.scene.Scene;
import javafx.scene.control.TabPane;
import javafx.stage.Stage;

public class MainGui extends Application {
    public static Stage stage;

    public final void main(String args) {
        Application.launch(args);
    }

    public final void initialiseDictionary() throws URISyntaxException {
//        ChineseDictionaryExtractor.readInDictionary();
    }

    public void start(Stage givenStage) throws URISyntaxException, IOException {
        stage = givenStage;
        this.initialiseDictionary();
        FXMLLoader loader = new FXMLLoader();
        InputStream fxmlStream = this.getClass().getResourceAsStream("/GUI/MainGUI.fxml");
        Object tabPane = loader.load(fxmlStream);

        TabPane root = (TabPane) tabPane;
        Scene scene = new Scene(root);
        givenStage.setScene(scene);
        givenStage.setTitle("HanziToAnki");
        givenStage.show();
    }

    public final Stage getStage() {
        return stage;
    }

    public final void setStage(Stage newStage) {
        stage = newStage;
    }
}

File: HanziToAnki/src/main/java/MainGUI.kt
package HanziToAnki

import javafx.application.Application
import javafx.scene.Scene
import javafx.stage.Stage
import java.io.FileInputStream
import javafx.fxml.FXMLLoader
import javafx.scene.control.TabPane

class MainGUI(): Application() {
    companion object {//may not be the best solution... but this treats it like a Java static variable
        @JvmStatic var stage: Stage? = null
    }
    fun main(args:String){
        launch(args)
    }
    fun initialiseDictionary(){
        Extract.readInDictionary()//Possibly not the best place for this
    }
    override fun start(givenStage: Stage){
        stage=givenStage
        initialiseDictionary()
        val loader: FXMLLoader = FXMLLoader()
        val fxmlDocPath:String = "../GUIs/TabbedUI.fxml"
        val fxmlStream: FileInputStream = FileInputStream(fxmlDocPath)
        val root: TabPane = loader.load(fxmlStream) as TabPane
        val scene: Scene = Scene(root)
        givenStage.setScene(scene)
        givenStage.setTitle("HanziToAnki")
        givenStage.show()
    }
}

/**
 * Created by james on 07/06/17.
 */
